src/python/pants/backend/awslambda/python/target_types_test.py

```diff
@@ -74,51 +74,83 @@ def test_handler_filespec(handler: str, expected: List[str]) -> None:
 
 
 def test_resolve_handler(rule_runner: RuleRunner) -> None:
-    def assert_resolved(handler: str, *, expected: str) -> None:
+    def assert_resolved(handler: str, *, expected: str, is_file: bool) -> None:
         addr = Address("src/python/project")
-        rule_runner.create_file("src/python/project/lambda.py")
-        rule_runner.create_file("src/python/project/f2.py")
+        rule_runner.write_files(
+            {"src/python/project/lambda.py": "", "src/python/project/f2.py": ""}
+        )
         field = PythonAwsLambdaHandlerField(handler, addr)
         result = rule_runner.request(
             ResolvedPythonAwsHandler, [ResolvePythonAwsHandlerRequest(field)]
         )
         assert result.val == expected
+        assert result.file_name_used == is_file
 
-    assert_resolved("path.to.lambda:func", expected="path.to.lambda:func")
-    assert_resolved("lambda.py:func", expected="project.lambda:func")
+    assert_resolved("path.to.lambda:func", expected="path.to.lambda:func", is_file=False)
+    assert_resolved("lambda.py:func", expected="project.lambda:func", is_file=True)
 
     with pytest.raises(ExecutionError):
-        assert_resolved("doesnt_exist.py:func", expected="doesnt matter")
+        assert_resolved("doesnt_exist.py:func", expected="doesnt matter", is_file=True)
     # Resolving >1 file is an error.
     with pytest.raises(ExecutionError):
-        assert_resolved("*.py:func", expected="doesnt matter")
+        assert_resolved("*.py:func", expected="doesnt matter", is_file=True)
 
 
 def test_inject_handler_dependency(rule_runner: RuleRunner, caplog) -> None:
-    rule_runner.add_to_build_file(
```

Comment:
Rewritten to use `.write_files()`. Only change made is adding `ambiguous_in_another_root.py` file and targets.